cmd/oras/internal/parse/parse.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parse
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+)
+
+var ErrMediaTypeNotFound = errors.New(`media type is not specified via the flag "--media-type" nor in the manifest.json`)
+
+// MediaTypeFromJson parses the media type field of bytes content in json format.
+func MediaTypeFromJson(cmd *cobra.Command, content []byte) (string, error) {
```

Comment:
`MediaTypeFromJson` should not assembles the `Error` struct. `manifest.json` is not hard-coded. The value should be read from the `fileRef` argument of `oras manifest push`

Reply:
moved the error to `manifest push` level. Added the function argument for filename.